### PR TITLE
Add unit tests using fake client

### PR DESCRIPTION
**Pull request by komalsukhani**
## Description

Added unit tests using [fake client](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake)

## Related Issue

[TT-5177](https://tyktech.atlassian.net/browse/TT-5177)

## Motivation and Context

Improve unit test coverage of the project

## Test Coverage For This Change

Added unit tests 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](https://github.com/TykTechnologies/tyk-operator/blob/master/CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] If you've changed API models, please update CRDs.
  - [ ] `make manifests`
  - [ ] `make helm`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `gofmt -s -w .`
  - [x] `go vet ./...`
  - [x] `golangci-lint run`

### CODE REVIEW

File: controllers/ingress_controller_test.go [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+	"github.com/matryer/is"
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
 
 func TestShortHash(t *testing.T) {
 	if expect, got := "c09b5928e", shortHash("foo.com"+"/httpbin22222"); got != expect {
```
Comment (buraksekili) [inline controllers/ingress_controller_test.go:18]: I know that this test case does not belong to the PR, but we can add more "unhappy" cases here. For example, what happens if domain or host names are empty?

File: controllers/operatorcontext_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+func TestOperatorContextDelete(t *testing.T) {
+	is := is.New(t)
+	dummyName := "dummy"
+
+	tests := map[string]struct {
+		Name        string
+		OpCtxStatus *v1alpha1.OperatorContextStatus
+		ReturnError bool
```
Comment (buraksekili) [inline controllers/operatorcontext_controller_test.go:69]: Considering this is the initial work for unit tests, it looks good to me. But what do you think about checking for errors in `error` type instead of `bool` in the future? So, we can understand that we obtained the desired error instead of any errors.

**Review by buraksekili — APPROVED** [review]
lgtm!

File: controllers/operatorcontext_controller_test.go (hunk 2 above)
Comment (komalsukhani) [reply controllers/operatorcontext_controller_test.go]: Updated tests.

File: controllers/ingress_controller_test.go (hunk 1 above)
Comment (komalsukhani) [reply controllers/ingress_controller_test.go:18]: Yes. We should do that.
